### PR TITLE
pull: follow redirects when checking if bottle is published

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

The error I was getting:
```
==> Verifying bottles published on Bintray
Verifying bottle: wpscan-2.9.3.x86_64_linux.bottle.tar.gz
Error: Failed to find published wpscan bottle at https://linuxbrew.bintray.com/bottles/wpscan-2.9.3.x86_64_linux.bottle.tar.gz (302 )!
```

Following the redirection fixes that.

### CODE REVIEW

**Comment by iMichka** [comment]
cc @sjackman @reitermarkus @jonchang @maxim-belkin

File: Library/Homebrew/dev-cmd/pull.rb [hunk 1]
```diff
@@ -605,6 +593,37 @@ def verify_bintray_published(formulae_names)
     end
   end
 
+  def check_publication(f, url, max_retries, retry_count)
+    http = create_http(url)
+    http.start do
+      res = get_http_result(http, url)
+      return true if res.is_a?(Net::HTTPSuccess)
+      if res.is_a?(Net::HTTPRedirection)
+        retry_count += 1
```
Comment (maxim-belkin) [inline Library/Homebrew/dev-cmd/pull.rb]: I'd call it `redirect_count`.

Comment (iMichka) [reply Library/Homebrew/dev-cmd/pull.rb]: I used the existing `retry_count` variable, which is used on the main loop, so I kept the same name for consistency.

In fact I could have a separate `retry_count` and `redirect_count` variable, and have 2 different max retries/redirects, but I am not sure it is worth it. Max values of retries is 300, and I do not expect more than 5 redirects (though I am not a web developer, so I may be wrong).

Comment (maxim-belkin) [reply Library/Homebrew/dev-cmd/pull.rb]: Yes, I noticed the resuse... while I agree it's safe to expect just a few redirects, imagine every retry taking 10 redirects. This way you reduce the number of retries from 300 to 30.

File: Library/Homebrew/dev-cmd/pull.rb [hunk 2]
```diff
@@ -605,6 +593,37 @@ def verify_bintray_published(formulae_names)
     end
   end
 
+  def check_publication(f, url, max_retries, retry_count)
```
Comment (maxim-belkin) [inline Library/Homebrew/dev-cmd/pull.rb]: I'd call it `check_<something>_status` where `<something>` could be `bottle`

**Comment by maxim-belkin** [comment]
LGTM! :)

File: Library/Homebrew/dev-cmd/pull.rb (hunk 2 above)
Comment (iMichka) [reply Library/Homebrew/dev-cmd/pull.rb]: What about `is_bottle_published` (it returns a boolean)?

Comment (maxim-belkin) [reply Library/Homebrew/dev-cmd/pull.rb]: I expect someone from upstream to chime in. I just expressed my opinion :)

**Comment by MikeMcQuaid** [comment]
I'd like to verify this solves a macOS problem first.

**Comment by sjackman** [comment]
To recap a conversation in Slack… unpublished or very recently published bottles are not redirected with an HTTP 302 redirect, but older bottles are redirected. It's unclear when that transition happens.

**Comment by sjackman** [comment]
For example, this bottle published ten hours ago, is already being redirected:
```
❯❯❯ curl -sD- https://homebrew.bintray.com/bottles/libxml2-2.9.5.sierra.bottle.tar.gz | head -n1
HTTP/1.1 302
```

**Comment by iMichka** [comment]
I'm with killing that pull request. It does not solve any practical problem on mac, and I will adapt my workflow for Linux.

Just tell me if you still would like to have these changes or not.

**Comment by sjackman** [comment]
As @jcount discovered, though only HTTP `GET` are redirected. HTTP `HEAD` are not.
 ```
❯❯❯ curl -sI https://homebrew.bintray.com/bottles/libxml2-2.9.4_4.el_capitan.bottle.tar.gz | head -n1
HTTP/1.1 200 OK
```

**Comment by MikeMcQuaid** [comment]
> I'm with killing that pull request. It does not solve any practical problem on mac, and I will adapt my workflow for Linux.
> 
> Just tell me if you still would like to have these changes or not.

:cool:, passing on this for now, thanks anyway @iMichka!

**Comment by iMichka** [comment]
Thanks, no problem :) As it does not fix anything, no need to merge it :)

**Pull request closed without merging**